pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/FunctionsImpl.java

```diff
@@ -722,20 +581,101 @@ public void updateFunctionOnWorkerLeader(final String tenant,
         }
     }
 
-    private Function.FunctionDetails validateUpdateRequestParams(final String tenant,
-                                                                 final String namespace,
-                                                                 final String componentName,
-                                                                 final FunctionConfig functionConfig,
-                                                                 final File componentPackageFile) throws IOException {
+    private void requestParamsPreCheck(String tenant, String namespace, String functionName, FunctionConfig functionConfig) {
+        requestParamsPreCheck(tenant, namespace, functionName);
+        if (functionConfig == null) {
+            throw new RestException(Response.Status.BAD_REQUEST, "Function config is not provided");
+        }
+    }
+
+    private void requestParamsPreCheck(String tenant, String namespace, String functionName) {
+        if (tenant == null) {
+            throw new RestException(Response.Status.BAD_REQUEST, "Tenant is not provided");
+        }
+        if (namespace == null) {
+            throw new RestException(Response.Status.BAD_REQUEST, "Namespace is not provided");
+        }
+        if (functionName == null) {
+            throw new RestException(Response.Status.BAD_REQUEST, "Function name is not provided");
+        }
+    }
+
+    // validate the operating tenant exists and has proper permissions to operate
+    private void validateFunctionNamespaceOperation(String tenant, String namespace, String functionName,
```

Comment:
You reverse the order of the original code. The original code checks `isAuhtorizedRole` first. Please don't change any code when you refactor code.